src/app/services/recovery.service.ts

```diff
@@ -379,65 +385,55 @@ export class RecoveryService {
   }
 
   private getAddressData(address: any, coin: string, network: string, cb: Function): any {
-    // call insight API to get address information
-    this.checkAddress(address.addressObject, coin, network).then((respAddressObs: any) => {
-
-      respAddressObs.subscribe(respAddress => {
-        // call insight API to get utxo information
-        this.checkUtxos(address.addressObject, coin, network).then((respUtxo: any) => {
-          respUtxo.subscribe(respUtxoData => {
-
-            let addr = respAddress.addrStr;
-            if (coin === 'bch' && this.useCashAddr && respUtxoData.length) {
-              // this is to fix blockdozer 1xxx address response
-              addr = address.addressObject.toCashAddress().split(':')[1];
-              _.each(respUtxoData, (r) => {
-                r.address = addr;
-              });
-            }
-
-            const addressData = {
-              address: addr,
-              balance: respAddress.balance,
-              unconfirmedBalance: respAddress.unconfirmedBalance,
-              utxo: respUtxoData,
-              privKeys: address.privKeys,
-              pubKeys: address.pubKeys,
-              info: address.info,
-              index: address.index,
-              isActive: respAddress.unconfirmedTxApperances + respAddress.txApperances > 0,
-            };
-            // TODO: Review this comment
-            // $rootScope.$emit('progress', _.pick(addressData, 'info', 'address', 'isActive', 'balance'));
-
-            /* This timeout is because we must not exceed the limit of 30 requests per minute to the server.
-            If you do, you will get an HTTP 429 error */
-            setTimeout(() => {
-              if (addressData.isActive) {
-                return cb(null, addressData);
-              }
-              return cb();
-            }, 1000);
-          });
-        });
-      });
+    this.getAddressInfo(address.addressObject, coin, network).subscribe((res) => {
+      const addr = coin === 'bch' && this.useCashAddr
+        ? address.addressObject.toCashAddress().split(':')[1]
+        : address.addressObject.toString();
+
+      const addressData = {
+        address: addr,
+        balance: res.balance.balance,
+        unconfirmedBalance: res.balance.unconfirmed,
+        utxo: res.utxos,
+        privKeys: address.privKeys,
+        pubKeys: address.pubKeys,
+        info: address.info,
+        index: address.index,
+        isActive: res.balance.unconfirmed + res.balance.confirmed > 0,
```

Comment:
an address could be active, even if it has not  unconfirmed or confirmed balance.

You need to check spent TXO.

Reply:
Now is active if an addr has at least 1 txo